### PR TITLE
key cant be null and cache tostring

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs [hunk 1]
```diff
@@ -35,12 +35,12 @@ protected async Task<TestingPlatformClient> StartAsServerAndConnectToTheClientAs
         foreach (DictionaryEntry entry in Environment.GetEnvironmentVariables())
         {
             // Skip all unwanted environment variables.
-            if (WellKnownEnvironmentVariables.ToSkipEnvironmentVariables.Contains(entry.Key!.ToString(), StringComparer.OrdinalIgnoreCase))
+            if (WellKnownEnvironmentVariables.ToSkipEnvironmentVariables.Contains(entry.Key.ToString(), StringComparer.OrdinalIgnoreCase))
```
Comment (Evangelink) [inline test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs]:
```suggestion
            string? key = entry.Key.ToString();
            if (WellKnownEnvironmentVariables.ToSkipEnvironmentVariables.Contains(entry.Key.ToString(), StringComparer.OrdinalIgnoreCase))
```

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs [hunk 2]
```diff
@@ -35,12 +35,12 @@ protected async Task<TestingPlatformClient> StartAsServerAndConnectToTheClientAs
...
             {
                 continue;
             }
 
-            environmentVariables[entry.Key!.ToString()!] = entry.Value!.ToString()!;
+            environmentVariables[entry.Key.ToString()!] = entry.Value!.ToString()!;
```
Comment (Evangelink) [inline test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs]:
```suggestion
            environmentVariables[key] = entry.Value!.ToString()!;
```

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs [hunk 3]
```diff
@@ -83,12 +83,12 @@ protected async Task<TestingPlatformClient> StartAsServerAndConnectAsync(TestHos
         foreach (DictionaryEntry entry in Environment.GetEnvironmentVariables())
         {
             // Skip all unwanted environment variables.
-            if (WellKnownEnvironmentVariables.ToSkipEnvironmentVariables.Contains(entry.Key!.ToString(), StringComparer.OrdinalIgnoreCase))
+            if (WellKnownEnvironmentVariables.ToSkipEnvironmentVariables.Contains(entry.Key.ToString(), StringComparer.OrdinalIgnoreCase))
```
Comment (Evangelink) [inline test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs]:
```suggestion
            string? key = entry.Key.ToString();
            if (WellKnownEnvironmentVariables.ToSkipEnvironmentVariables.Contains(entry.Key.ToString(), StringComparer.OrdinalIgnoreCase))
```

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs [hunk 4]
```diff
@@ -83,12 +83,12 @@ protected async Task<TestingPlatformClient> StartAsServerAndConnectAsync(TestHos
...
             {
                 continue;
             }
 
-            environmentVariables[entry.Key!.ToString()!] = entry.Value!.ToString()!;
+            environmentVariables[entry.Key.ToString()!] = entry.Value!.ToString()!;
```
Comment (Evangelink) [inline test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs]:
```suggestion
            environmentVariables[key] = entry.Value!.ToString()!;
```

**Review by Evangelink — COMMENTED** [review]
You have missed some caching of keys, corrected from GH so might be not 100% correct.

**Comment by SimonCropp** [comment]
@Evangelink side note: feel free at any time to push commits directly to my PR branches

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs [hunk 5]
```diff
@@ -35,12 +35,13 @@ protected async Task<TestingPlatformClient> StartAsServerAndConnectToTheClientAs
         foreach (DictionaryEntry entry in Environment.GetEnvironmentVariables())
         {
             // Skip all unwanted environment variables.
-            if (WellKnownEnvironmentVariables.ToSkipEnvironmentVariables.Contains(entry.Key!.ToString(), StringComparer.OrdinalIgnoreCase))
+            string? key = entry.Key.ToString();
+            if (WellKnownEnvironmentVariables.ToSkipEnvironmentVariables.Contains(entry.Key.ToString(), StringComparer.OrdinalIgnoreCase))
```
Comment (cremor) [inline test/IntegrationTests/MSTest.Acceptance.IntegrationTests/ServerMode/ServerModeTestsBase.cs]: Shouldn't this line now use `key` as parameter for `Contains`?
Same for the similar call below.

**Review by Evangelink — APPROVED** [review]

**Pull request merged**